crates/solidity/inputs/language/bindings/rules.msgb

```diff
@@ -349,6 +378,12 @@ inherit .enclosing_def
     )]
 ]] {
   edge @contract.lexical_scope -> @member.def
+
+  ;; Special case: for the built-ins file, export state variables in the
```

Comment:
This will export all members (functions, structs, etc...), not just state variables, right?

Reply:
This was needed because normally contract state variables are not accessible externally, and we need that to be able to emulate globals (eg. the global `tx` is linked in the stack graph as `$BuiltIns$.tx`). Functions and structs are exposed in a different scope `type_members` for this reason.

But, as I said in the previous reply, I think there's a better way that would be less confusing.

Reply:
> This was needed because normally contract state variables are not accessible externally, 

They are available externally, just via a function syntax, right?

```solidity
contract A {
    uint public data;
}
function foo(A a) view returns (uint) {
    return a.data();                          // bound to `A::data`
}
```

Do we already bound them as function/callable properties? do we try to bind only the identifier if not a call syntax `a.data`?